reflection/walk_test.go

```diff
@@ -0,0 +1,166 @@
+package reflection
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+type Person struct {
+	Name    string
+	Profile Profile
+}
+
+type Profile struct {
+	Age  int
+	City string
+}
+
+func TestWalk(t *testing.T) {
+	assert := require.New(t)
+	cases := []struct {
```

Comment:
Why not create this struct outside the TestWalk function?

Reply:
Thanks Nacho!

It's there because is only used under the scope of that test, as base for the table testing approach.

Does it make sense @JuanCalcagno416 ?

Reply:
True, now that I see the rest of the structs, you can have a base struct and now that you are playing with reflection, you could edit the field you need for each test and no need to create a struct. Of course, this is just overcomplicating stuff, reflection can be confusing XD

Reply:
Thanks @JuanCalcagno416, however I will pass on overengineering :-P